Docs/source/building/summit.rst

```diff
@@ -103,6 +108,22 @@ Optionally, download and build openPMD-api for I/O:
    cmake ../openPMD-api -DopenPMD_USE_PYTHON=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/openPMD-api-install/ -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_RPATH='$ORIGIN' -DMPIEXEC_EXECUTABLE=$(which jsrun)
    cmake --build . --target install --parallel 16
 
+Optionally, download and install :ref:`libEnsemble <libensemble>` for dynamic ensemble optimizations:
+
+.. code-block:: bash
+
+   export BLAS=$OLCF_OPENBLAS_ROOT/lib/libopenblas.so
+   export LAPACK=$OLCF_NETLIB_LAPACK_ROOT/lib64/liblapack.so
+   python3 -m pip install --user --upgrade pip
+   python3 -m pip install --user virtualenv
+   python3 -m venv $HOME/sw/venvs/warpx-libE
+   source $HOME/sw/venvs/warpx-libE/bin/activate
+   python3 -m pip install --upgrade pip
+   python3 -m pip install --upgrade cython
+   python3 -m pip install --upgrade numpy
+   python3 -m pip install --upgrade scipy
+   python3 -m pip install --upgrade -r $HOME/src/warpx/Tools/LibEnsemble/requirements.txt
+
```

Comment:
Thanks for all the help and this PR again @ax3l !

I managed to make the libEnsemble test run in the WarpX repo work and I followed their repo's documentation in the installation:

```bash
module load python
conda create --name LDA_conda python=3.7
export PYTHONNOUSERSITE=1
source activate LDA_conda
module load gcc
module load cuda
CC=mpicc MPICC=mpicc pip install mpi4py --no-binary mpi4py
pip install libensemble
conda install -n LDA_conda -c conda-forge matplotlib
conda install -n LDA_conda -c conda-forge numpy
conda install -n LDA_conda -c conda-forge scipy
conda install -n LDA_conda -c conda-forge yt
conda install -n LDA_conda -c conda-forge nlopt
conda install -n LDA_conda -c conda-forge petsc4py
pip install DFO-LS
conda deactivate
```

And so I was going to ask if you want to "merge" this into your instructions here.

And add `petsc4py` to the requirements.txt file.

I can also do it in another PR in the docs part for libEnsemble.

Thanks

Reply:
I would avoid mixing this with conda, since it will make re-compiles of WarpX potentially problematic.

Reply:
@MaxThevenet did we need `mpi4py` and `petsc4py` even in those scripts? I think not last time.

Reply:
Hi @ax3l ! Sorry, this comment was outdated. With the 2 PRs of yesterday by @MaxThevenet and @atmyers we no longer get error messages when the `petsc4py` package is missing.
So you can disregard this comment.

Reply:
I'll add the `mpi4py` example as a comment, since it's generally useful for the future.